`@targetName` is only checked in non-primary constructors

## Compiler version

abfb3f99ab88e2bd4b4d57807ca4d44d0632cd47

## Minimized code

```scala
import scala.annotation.targetName

class Foo @targetName("bla") ()
```

## Output

```scala
Exception in thread "main" 
  unhandled exception while running erasure on test.scala

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose
  For non-enriched exceptions, compile with -Xno-enrich-error-messages.


     while compiling: test.scala
        during phase: erasure
                mode: Mode(ImplicitsEnabled)
     library version: version 3.8.1-RC1-bin-SNAPSHOT-nonbootstrapped
    compiler version: version 3.8.1-RC1-bin-SNAPSHOT-nonbootstrapped-git-abfb3f9
            settings: -Vprint List(typer) -classpath <redacted>
java.lang.AssertionError: assertion failed

        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:13)
        at dotty.tools.dotc.core.Contexts$FreshContext.setOwner(Contexts.scala:647)
        at dotty.tools.dotc.core.Contexts$Context.superOrThisCallContext(Contexts.scala:446)
        at dotty.tools.dotc.core.Contexts$Context.superCallContext(Contexts.scala:424)
        at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:3219)
        at dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1048)
        at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3753)
        at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3757)
        at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3858)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3946)
        at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3951)
```

## Expectation

Should not compile and obviously not crash the compiler.

## Notes

I noticed this while reading the code of `Typer` that we only do the check for non-primary constructors :

https://github.com/scala/scala3/blob/abfb3f99ab88e2bd4b4d57807ca4d44d0632cd47/compiler/src/dotty/tools/dotc/typer/Typer.scala#L3126-L3140

--- a/compiler/src/dotty/tools/dotc/typer/Typer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Typer.scala
@@ -3126,6 +3126,10 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
     if sym.isConstructor then
       if sym.is(Inline) then
         report.error("constructors cannot be `inline`", ddef)
+
+      if sym.targetName != sym.name then
+        report.error(em"@targetName annotation may not be used on a constructor", ddef.srcPos)
+
       if sym.isPrimaryConstructor then
         if sym.owner.is(Case) then
           for
@@ -3135,9 +3139,6 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
             if defn.isContextFunctionType(param.tpt.tpe) then
               report.error("case class element cannot be a context function", param.srcPos)
       else
-        if sym.targetName != sym.name then
-          report.error(em"@targetName annotation may not be used on a constructor", ddef.srcPos)
-
         for params <- paramss1; param <- params do
           checkRefsLegal(param, sym.owner, (name, sym) => sym.is(TypeParam), "secondary constructor")
 

--- a/tests/neg/i24681.check
+++ b/tests/neg/i24681.check
@@ -1,0 +1,4 @@
+-- Error: tests/neg/i24681.scala:3:9 -----------------------------------------------------------------------------------
+3 |class Foo @targetName("bla") () // error
+  |         ^^^^^^^^^^^^^^^^^^^^^^
+  |         @targetName annotation may not be used on a constructor

--- a/tests/neg/i24681.scala
+++ b/tests/neg/i24681.scala
@@ -1,0 +1,3 @@
+import scala.annotation.targetName
+
+class Foo @targetName("bla") () // error

